src/komponenter/kalkulator/Uregningskolonne/Utregningskolonne.tsx

```diff
@@ -19,36 +22,56 @@ interface UtregningskolonneProps {
 const Utregningskolonne: FunctionComponent<UtregningskolonneProps> = (
     props
 ) => {
-    const overSiktOverPerioderInnenfor18mnd: AllePermitteringerOgFraværesPerioder = {
-        andreFraværsperioder:
-            props.allePermitteringerOgFraværesPerioder.andreFraværsperioder,
+    const [
+        oversiktOverPerioderInnenfor18mnd,
+        setOversiktOverPerioderInnenfor18mnd,
+    ] = useState<AllePermitteringerOgFraværesPerioder>({
+        andreFraværsperioder: [],
         permitteringer: [],
-    };
-    props.allePermitteringerOgFraværesPerioder.permitteringer.forEach(
-        (periode) => {
-            const kuttetDatoIntervall = kuttAvDatoIntervallInnefor18mnd(
-                periode,
-                finnDato18MndTilbake(props.sisteDagIPeriode),
-                props.sisteDagIPeriode
-            );
-            overSiktOverPerioderInnenfor18mnd.permitteringer.push(
-                kuttetDatoIntervall
-            );
-        }
-    );
-    const oversiktOverDager: OversiktOverBrukteOgGjenværendeDager = sumPermitteringerOgFravær(
-        overSiktOverPerioderInnenfor18mnd,
-        props.sisteDagIPeriode
-    );
+    });
+
+    const [
+        resultatUtregningAv18mndsPeriode,
+        setResultatUtregningAv18mndsPeriode,
+    ] = useState<OversiktOverBrukteOgGjenværendeDager>({
+        dagerPermittert: 0,
+        dagerAnnetFravær: 0,
+        dagerGjensående: 0,
+    });
+
+    useEffect(() => {
+        const avkuttet18mndPerioder: AllePermitteringerOgFraværesPerioder = {
+            andreFraværsperioder:
+                props.allePermitteringerOgFraværesPerioder.andreFraværsperioder,
+            permitteringer: [],
+        };
+        props.allePermitteringerOgFraværesPerioder.permitteringer.forEach(
+            (periode) => {
+                const kuttetDatoIntervall: DatoIntervall = kuttAvDatoIntervallInnefor18mnd(
+                    periode,
+                    finnDato18MndTilbake(props.sisteDagIPeriode),
+                    props.sisteDagIPeriode
+                );
+                avkuttet18mndPerioder.permitteringer.push(kuttetDatoIntervall);
+            }
+        );
+        setOversiktOverPerioderInnenfor18mnd(avkuttet18mndPerioder);
+    }, [props.allePermitteringerOgFraværesPerioder, props.sisteDagIPeriode]);
+
+    useEffect(() => {
```

Comment:
Nå har vi en state som er avhengig av en annen (`resultatUtregningAv18mndsPeriode` er avhengig av `oversiktOverPerioderInnenfor18mnd`). Da trigges det en ekstra render. Hva med å skrive følgende i stedet for denne useEffecten?
```
const resultatUtregningAv18mndsPeriode = sumPermitteringerOgFravær(oversiktOverPerioderInnenfor18mnd)
```